src/VisualStudio/Core/Def/Packaging/PackageSearchService.cs

```diff
@@ -94,12 +91,49 @@ private static IPackageSearchRemoteControlService CreateRemoteControlService(VSS
 
             _cacheDirectoryInfo = new DirectoryInfo(Path.Combine(
                 localSettingsDirectory, "PackageCache", string.Format(Invariant($"Format{_dataFormatVersion}"))));
-            // _databaseFileInfo = new FileInfo(Path.Combine(_cacheDirectoryInfo.FullName, "NuGetCache.txt"));
 
             _cancellationTokenSource = cancellationTokenSource;
             _cancellationToken = _cancellationTokenSource.Token;
         }
 
+        internal void Start()
+        {
+            var options = _workspace.Options;
+            if (!options.GetOption(ServiceComponentOnOffOptions.PackageSearch))
+            {
+                return;
+            }
+
+            // Start the whole process once we're connected
+            _installerService.PackageSourcesChanged += OnPackageSourcesChanged;
+            OnPackageSourcesChanged(this, EventArgs.Empty);
+            _started = true;
+        }
+
+        internal void Stop()
+        {
+            if (!_started)
+            {
+                return;
+            }
+
+            _installerService.PackageSourcesChanged -= OnPackageSourcesChanged;
+            // Cancel any existing work.
+            _cancellationTokenSource.Cancel();
```

Comment:
No.  This type is a single "start -> stop" type.  It's not restartable.  _started coudl be renamed to "hasStartBeenCalled" to better convey that it's just to let us know if we're stopping after the initial start was called.